Identity Hub CLI (#30)
agera-edc/IdentityHub: system-tests/README.md

## System tests

> System tests run a sample EDC connector with the Identity Hub extension and a DID server with sample DID documents for the connector and an external authority using docker. The test checks the correct functioning of the Identity Hub CLI together with the CredentialsVerifier.

Reviewer: first sentence with multiple `and`/`with` clauses is hard to parse

Author: I reworded this section a bit. Please have a look.

> An instance of `CredentialsVerifiar` is injected in the test (hence the `@ExtendWith(EdcExtension.class)` annotation), to verify that another EDC connector is able to retrieve credentials from the Identity Hub after these have beed added using the CLI.

Reviewer: ```suggestion
An instance of `CredentialsVerifier` is injected in the test (hence the `@ExtendWith(EdcExtension.class)` annotation), to verify that another EDC connector is able to retrieve credentials from the Identity Hub after these have been added using the CLI.
```

#### Local test resources

> Following test resources are provided in order to run system tests:

Reviewer: ```suggestion
The following test resources are used to run system tests:
```

> - A set of Private and Public keys for both the external authority and the EDC connector (identity hub owner) e.g. `system-tests/resources/jwt/authority`.

Reviewer: ```suggestion
- A set of Private and Public keys for both the external authority and the EDC connector (identity hub owner) at `system-tests/resources/jwt/authority`.
```

>     ```bash

Reviewer: please explain what these commands are for (also so that the user understands they don't need to run them just to run the test)

    # generate a private key
    openssl ecparam -name prime256v1 -genkey -noout -out private-key.pem
    # generate corresponding public key
    openssl ec -in private-key.pem -pubout -out public-key.pem
    ```
- Web DIDs are available under `system-tests/resources/webdid` folder. The `publicKeyJwk` section of each `did.json` was generated by converting the corresponding public key to JWK format, for example the authority public key was converted to JWK using following command:

    ```bash
    docker run -i danedmunds/pem-to-jwk:1.2.1 --public --pretty < system-tests/tests/resources/jwt/participant/public-key.pem
    ```

## Running tests locally 

Build launcher for system tests

> ```bash

Reviewer: ```suggestion

```bash
```
